### PR TITLE
feat(QScrollArea): backport QScrollArea from v2 (both vertical and horizontal) #7471; save and restore scroll position in keep-alive

### PR DESCRIPTION
**Pull request by pdanpdan**

### CODE REVIEW

File: ui/src/components/scroll-area/QScrollArea.json [hunk 1]
```diff
@@ -11,19 +11,51 @@
...
+      "category": "style",
+      "addedIn": "v1.16.0"
+    },
+
+    "horizontal-thumb-style": {
+      "type": "Object",
+      "desc": "Object with CSS properties and values for custom styling the thumb of the horizontal scrollbar; Is applied on top of 'thumb-style' prop",
+      "examples": [ ":thumb-style=\"{ bottom: '4px', borderRadius: '5px', background: 'red', height: '10px', opacity: 1 }\"" ],
+      "category": "style",
+      "addedIn": "v1.16.0"
```
Comment (rstoenescu) [inline ui/src/components/scroll-area/QScrollArea.json]: v1.17 everywhere

File: ui/src/components/scroll-area/QScrollArea.js [hunk 2]
```diff
@@ -154,97 +244,162 @@ export default Vue.extend({
...
-        ? setHorizontalScrollPosition
-        : setScrollPosition
+    getScrollPercentage () {
+      return {
+        top: this.verticalScrollPercentage,
+        left: this.horizontalScrollPercentage
+      }
+    },
+
+    setScrollPosition (axis, offset, duration) {
```
Comment (rstoenescu) [inline ui/src/components/scroll-area/QScrollArea.js:294]: Put the `axis` as third parameter so we don't do breaking changes pls.

File: ui/src/components/scroll-area/QScrollArea.js [hunk 3]
```diff
@@ -154,97 +244,162 @@ export default Vue.extend({
...
+        info[ axis + 'Size' ] = data.size
+        info[ axis + 'ContainerSize' ] = this.container[ axis ]
+      })
+
+      return info
+    },
+
     getScrollPosition () {
-      return this.scrollPosition
+      return {
```
Comment (rstoenescu) [inline ui/src/components/scroll-area/QScrollArea.js]: Should return the same as in previous version. For non-breaking changes purposes, let's add a getVerticalScrollPosition() method as well, unless you have a better idea.

File: ui/src/components/scroll-area/QScrollArea.js [hunk 4]
```diff
@@ -154,97 +244,162 @@ export default Vue.extend({
...
+
+      const fn = axis === 'vertical'
+        ? setVerticalScrollPosition
+        : setHorizontalScrollPosition
 
       fn(this.$refs.target, offset, duration)
     },
 
-    setScrollPercentage (percentage, duration) {
+    setScrollPercentage (axis, percentage, duration) {
```
Comment (rstoenescu) [inline ui/src/components/scroll-area/QScrollArea.js:311]: For non-breaking changes purposes, let's add the `axis` param as third one.

**Review by rstoenescu — CHANGES_REQUESTED** [review]

File: ui/src/components/scroll-area/QScrollArea.js (hunk 2 above)
Comment (pdanpdan) [reply ui/src/components/scroll-area/QScrollArea.js:294]: I'll check first parameter if it's vertical/horizontal and change params

**Pull request merged**